bb-core/src/main/java/com/cognifide/qa/bb/loadable/hierarchy/WebElementInterceptor.java

```diff
@@ -57,29 +57,33 @@
   public Object invoke(MethodInvocation methodInvocation) throws Throwable {
     Class methodCallerClass = getMethodCallerClassWithoutGuiceContext();
 
-    if (methodCallerClass.isAnnotationPresent(PageObject.class)) {
+    if (isApplicable(methodCallerClass)) {
 
-      LOG.
-        debug("Caught invocation of method {} from {}. Started processing loadable component conditions hierarchy",
+      LOG.debug(
+          "Caught invocation of method {} from {}. Started processing loadable component conditions hierarchy",
           methodInvocation.getMethod().getName(), methodCallerClass.getName());
 
       BobcatWebElement caller = (BobcatWebElement) methodInvocation.getThis();
       ClassFieldContext directContext = acquireDirectContext(caller);
-      ConditionStack loadableContextHierarchy
-        = loadableCondsExplorer.
-        discoverLoadableContextHierarchy(directContext, pageObjectInvocationTracker.
-          getSubjectStack());
+      ConditionStack loadableContextHierarchy =
+          loadableCondsExplorer.discoverLoadableContextHierarchy(directContext,
+              pageObjectInvocationTracker.getSubjectStack());
 
       loadConditionChainRunner.chainCheck(loadableContextHierarchy);
     }
     return methodInvocation.proceed();
 
   }
 
+  private boolean isApplicable(Class methodCallerClass) {
+    return methodCallerClass.isAnnotationPresent(PageObject.class)
+        || LoadableComponentsUtil.isStepsImplementationClass(methodCallerClass)
```

Comment:
this is a very good point. I'll try to separate this logic and move it to bobcumber, and bb-junit modules